### PR TITLE
[201811][Devices] Add new device CIG CS6436-56P

### PR DESCRIPTION
**Pull request by michealylj1**
New device of CIG CS6436-56P :
CPU : Intel Atom E3940
ASIC Vendor: Nephos
Swich ASIC: NP8366
Port Configuration: 25GX48,100GX8
SONiC Image: sonic-nephos.bin

What I did
Add new device and platform files for CIG CS6436-56P device
How I did it
How to verify it
show interfaces status
show interfaces transceiver presence
show interfaces transceiver eeprom
show platform syseeprom
show platform psustatus

Signed-off-by: yanglijie@cigtech.com

### CODE REVIEW

File: device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# name          lanes             alias             index
```
Comment (zhenggen-xu) [inline device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini]: Add speed column?

File: device/cig/x86_64-cig_cs6436_56p-r0/minigraph.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<DeviceMiniGraph xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns="Microsoft.Search.Autopilot.Evolution">
```
Comment (zhenggen-xu) [inline device/cig/x86_64-cig_cs6436_56p-r0/minigraph.xml]: minigraph.xml is not needed.

File: device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini (hunk 1 above)
Comment (michealylj1) [reply device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini]: I just updateit .Thanks

File: device/cig/x86_64-cig_cs6436_56p-r0/minigraph.xml (hunk 2 above)
Comment (michealylj1) [reply device/cig/x86_64-cig_cs6436_56p-r0/minigraph.xml]: already delete it. Thanks

File: device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini (hunk 1 above)
Comment (michealylj1) [reply device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini]: The output of command "show interface status" as:

root@switch1:~# show interface status             
  Interface            Lanes    Speed    MTU         Alias    Oper    Admin    Type    Asym PFC
-----------  ---------------  -------  -----  ------------  ------  -------  ------  ----------
  Ethernet0                8      25G   9100   Ethernet1/1      up       up     SFP         N/A
  Ethernet1                9      25G   9100   Ethernet2/1      up       up     SFP         N/A
  Ethernet2               10      25G   9100   Ethernet3/1    down     down     N/A         N/A
  Ethernet3               11      25G   9100   Ethernet4/1    down     down     N/A         N/A
  Ethernet4               12      25G   9100   Ethernet5/1    down     down     N/A         N/A
  Ethernet5               13      25G   9100   Ethernet6/1    down     down     N/A         N/A
  Ethernet6               14      25G   9100   Ethernet7/1    down     down     N/A         N/A
  Ethernet7               15      25G   9100   Ethernet8/1    down     down     N/A         N/A
  Ethernet8               16      25G   9100   Ethernet9/1    down     down     N/A         N/A
  Ethernet9               17      25G   9100  Ethernet10/1    down     down     N/A         N/A
 Ethernet10               18      25G   9100  Ethernet11/1    down     down     N/A         N/A
 Ethernet11               19      25G   9100  Ethernet12/1    down     down     N/A         N/A
 Ethernet12               20      25G   9100  Ethernet13/1    down     down     N/A         N/A
 Ethernet13               21      25G   9100  Ethernet14/1    down     down     N/A         N/A
 Ethernet14               22      25G   9100  Ethernet15/1    down     down     N/A         N/A
 Ethernet15               23      25G   9100  Ethernet16/1    down     down     N/A         N/A
 Ethernet16               32      25G   9100  Ethernet17/1    down     down     N/A         N/A
 Ethernet17               33      25G   9100  Ethernet18/1    down     down     N/A         N/A
 Ethernet18               34      25G   9100  Ethernet19/1    down     down     N/A         N/A
 Ethernet19               35      25G   9100  Ethernet20/1    down     down     N/A         N/A
 Ethernet20               40      25G   9100  Ethernet21/1    down     down     N/A         N/A
 Ethernet21               41      25G   9100  Ethernet22/1    down     down     N/A         N/A
 Ethernet22               42      25G   9100  Ethernet23/1    down     down     N/A         N/A
 Ethernet23               43      25G   9100  Ethernet24/1    down     down     N/A         N/A
 Ethernet24               48      25G   9100  Ethernet25/1    down     down     N/A         N/A
 Ethernet25               49      25G   9100  Ethernet26/1    down     down     N/A         N/A
 Ethernet26               50      25G   9100  Ethernet27/1    down     down     N/A         N/A
 Ethernet27               51      25G   9100  Ethernet28/1    down     down     N/A         N/A
 Ethernet28               56      25G   9100  Ethernet29/1    down     down     N/A         N/A
 Ethernet29               57      25G   9100  Ethernet30/1    down     down     N/A         N/A
 Ethernet30               58      25G   9100  Ethernet31/1    down     down     N/A         N/A
 Ethernet31               59      25G   9100  Ethernet32/1    down     down     N/A         N/A
 Ethernet32               64      25G   9100  Ethernet33/1    down     down     N/A         N/A
 Ethernet33               65      25G   9100  Ethernet34/1    down     down     N/A         N/A
 Ethernet34               66      25G   9100  Ethernet35/1    down     down     N/A         N/A
 Ethernet35               67      25G   9100  Ethernet36/1    down     down     N/A         N/A
 Ethernet36               68      25G   9100  Ethernet37/1    down     down     N/A         N/A
 Ethernet37               69      25G   9100  Ethernet38/1    down     down     N/A         N/A
 Ethernet38               70      25G   9100  Ethernet39/1    down     down     N/A         N/A
 Ethernet39               71      25G   9100  Ethernet40/1    down     down     N/A         N/A
 Ethernet40               72      25G   9100  Ethernet41/1    down     down     N/A         N/A
 Ethernet41               73      25G   9100  Ethernet42/1    down     down     N/A         N/A
 Ethernet42               74      25G   9100  Ethernet43/1    down     down     N/A         N/A
 Ethernet43               75      25G   9100  Ethernet44/1    down     down     N/A         N/A
 Ethernet44               76      25G   9100  Ethernet45/1    down     down     N/A         N/A
 Ethernet45               77      25G   9100  Ethernet46/1    down     down     N/A         N/A
 Ethernet46               78      25G   9100  Ethernet47/1    down     down     N/A         N/A
 Ethernet47               79      25G   9100  Ethernet48/1    down     down     N/A         N/A
 Ethernet48      84,85,86,87     100G   9100  Ethernet49/1      up       up  QSFP28         N/A
 Ethernet49      80,81,82,83     100G   9100  Ethernet50/1      up       up  QSFP28         N/A
 Ethernet50      92,93,94,95     100G   9100  Ethernet51/1    down     down     N/A         N/A
 Ethernet51      88,89,90,91     100G   9100  Ethernet52/1    down     down     N/A         N/A
 Ethernet52  108,109,110,111     100G   9100  Ethernet53/1    down     down     N/A         N/A
 Ethernet53  104,105,106,107     100G   9100  Ethernet54/1    down     down     N/A         N/A
 Ethernet54  116,117,118,119     100G   9100  Ethernet55/1    down     down     N/A         N/A
 Ethernet55  112,113,114,115     100G   9100  Ethernet56/1    down     down     N/A         N/A
root@switch1:~#

File: platform/nephos/sonic-platform-modules-cig/cs6436-56p/modules/x86-64-cig-cs6436-56p-sfp.c [hunk 3]
```diff
@@ -0,0 +1,1454 @@
...
+
+enum driver_type_e {
+	DRIVER_TYPE_SFP_MSA,
+	DRIVER_TYPE_SFP_DDM,
+	DRIVER_TYPE_QSFP
+};
+
+/* Each client has this additional data
+ */
+struct eeprom_data {
```
Comment (zhenggen-xu) [inline platform/nephos/sonic-platform-modules-cig/cs6436-56p/modules/x86-64-cig-cs6436-56p-sfp.c]: indentation for these structs looked odd, is it possible to fix them?

File: device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini (hunk 1 above)
Comment (zhenggen-xu) [reply device/cig/x86_64-cig_cs6436_56p-r0/Cig-CS6436-56P/port_config.ini]: how about these commands?
show interfaces transceiver presence
show interfaces transceiver eeprom
show platform syseeprom
show platform psustatus

File: device/cig/x86_64-cig_cs6436_56p-r0/plugins/sfputil.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        #toggle reset
+        reg_file.seek(0)
+        reg_file.write('1')
+        time.sleep(1)
+        reg_file.seek(0)
+        reg_file.write('0')
+        reg_file.close()
+        return True
+
+    def set_low_power_mode(self, port_nuM, lpmode):
```
Comment (zhenggen-xu) [inline device/cig/x86_64-cig_cs6436_56p-r0/plugins/sfputil.py]: What is the default lpmode setting?   Can you bring up optical ports on this platform?

Comment (michealylj1) [reply device/cig/x86_64-cig_cs6436_56p-r0/plugins/sfputil.py]: It's just defined one function interface, not any implement. So the optical ports in our devcie are on the normal mode.

Comment (michealylj1) [reply device/cig/x86_64-cig_cs6436_56p-r0/plugins/sfputil.py]: It's just defined one function interface, not any implement. So the optical ports in our devcie are on the normal mode.

File: platform/nephos/sonic-platform-modules-cig/cs6436-56p/modules/x86-64-cig-cs6436-56p-sfp.c (hunk 3 above)
Comment (michealylj1) [reply platform/nephos/sonic-platform-modules-cig/cs6436-56p/modules/x86-64-cig-cs6436-56p-sfp.c]: The code is similar as the Linux optoe driver "drivers/misc/eeprom/eeprom.c": 

/* Each client has this additional data */
struct eeprom_data {
        struct mutex update_lock;
        u8 valid;                       /* bitfield, bit!=0 if slice is valid */
        unsigned long last_updated[8];  /* In jiffies, 8 slices */
        u8 data[EEPROM_SIZE];           /* Register values */
        enum eeprom_nature nature;
};

**Comment by zhenggen-xu** [comment]
As mentioned, can you paste the commands output in the PR?
show interfaces transceiver presence
show interfaces transceiver eeprom
show platform syseeprom
show platform psustatus

**Comment by michealylj1** [comment]
root@switch1:~# show interfaces transceiver presence
Port        Presence
----------  -----------
Ethernet0   Present
Ethernet1   Present
Ethernet2   Not present
Ethernet3   Not present
Ethernet4   Not present
Ethernet5   Not present
Ethernet6   Not present
Ethernet7   Not present
Ethernet8   Not present
Ethernet9   Not present
Ethernet10  Not present
Ethernet11  Not present
Ethernet12  Not present
Ethernet13  Not present
Ethernet14  Not present
Ethernet15  Not present
Ethernet16  Not present
Ethernet17  Not present
Ethernet18  Not present
Ethernet19  Not present
Ethernet20  Not present
Ethernet21  Not present
Ethernet22  Not present
Ethernet23  Not present
Ethernet24  Not present
Ethernet25  Not present
Ethernet26  Not present
Ethernet27  Not present
Ethernet28  Not present
Ethernet29  Not present
Ethernet30  Not present
Ethernet31  Not present
Ethernet32  Not present
Ethernet33  Not present
Ethernet34  Not present
Ethernet35  Not present
Ethernet36  Not present
Ethernet37  Not present
Ethernet38  Not present
Ethernet39  Not present
Ethernet40  Not present
Ethernet41  Not present
Ethernet42  Not present
Ethernet43  Not present
Ethernet44  Not present
Ethernet45  Not present
Ethernet46  Not present
Ethernet47  Not present
Ethernet48  Present
Ethernet49  Present
Ethernet50  Not present
Ethernet51  Not present
Ethernet52  Not present
Ethernet53  Not present
Ethernet54  Not present
Ethernet55  Not present
root@switch1:~# 
root@switch1:~# show interfaces transceiver eeprom
Ethernet0: SFP EEPROM detected
        Connector: CopperPigtail
        EncodingCodes: Unspecified
        ExtIdentOfTypeOfTransceiver: GBIC/SFP defined by twowire interface ID
        LengthCable(UnitsOfm): 1
        NominalSignallingRate(UnitsOf100Mbd): 255
        RateIdentifier: Unspecified
        ReceivedPowerMeasurementType: OMA
        TransceiverCodes:
                SFP+CableTechnology: Passive Cable
        TypeOfTransceiver: SFP or SFP Plus
        VendorDataCode(YYYY-MM-DD Lot): 2018-11-16 
        VendorName: OEM
        VendorOUI: 00-02-c9
        VendorPN: SFP28-25G-CU0.5M
        VendorRev: 02
        VendorSN: 1811160050

Ethernet1: SFP EEPROM detected
        Connector: CopperPigtail
        EncodingCodes: Unspecified
        ExtIdentOfTypeOfTransceiver: GBIC/SFP defined by twowire interface ID
        LengthCable(UnitsOfm): 1
        NominalSignallingRate(UnitsOf100Mbd): 255
        RateIdentifier: Unspecified
        ReceivedPowerMeasurementType: OMA
        TransceiverCodes:
                SFP+CableTechnology: Passive Cable
        TypeOfTransceiver: SFP or SFP Plus
        VendorDataCode(YYYY-MM-DD Lot): 2018-11-16 
        VendorName: OEM
        VendorOUI: 00-02-c9
        VendorPN: SFP28-25G-CU0.5M
        VendorRev: 02
        VendorSN: 1811160050

Ethernet2: SFP EEPROM not detected

Ethernet3: SFP EEPROM not detected

Ethernet4: SFP EEPROM not detected

Ethernet5: SFP EEPROM not detected

Ethernet6: SFP EEPROM not detected

Ethernet7: SFP EEPROM not detected

Ethernet8: SFP EEPROM not detected

Ethernet9: SFP EEPROM not detected

Ethernet10: SFP EEPROM not detected

Ethernet11: SFP EEPROM not detected

Ethernet12: SFP EEPROM not detected

Ethernet13: SFP EEPROM not detected

Ethernet14: SFP EEPROM not detected

Ethernet15: SFP EEPROM not detected

Ethernet16: SFP EEPROM not detected

Ethernet17: SFP EEPROM not detected

Ethernet18: SFP EEPROM not detected

Ethernet19: SFP EEPROM not detected

Ethernet20: SFP EEPROM not detected

Ethernet21: SFP EEPROM not detected

Ethernet22: SFP EEPROM not detected

Ethernet23: SFP EEPROM not detected

Ethernet24: SFP EEPROM not detected

Ethernet25: SFP EEPROM not detected

Ethernet26: SFP EEPROM not detected

Ethernet27: SFP EEPROM not detected

Ethernet28: SFP EEPROM not detected

Ethernet29: SFP EEPROM not detected

Ethernet30: SFP EEPROM not detected

Ethernet31: SFP EEPROM not detected

Ethernet32: SFP EEPROM not detected

Ethernet33: SFP EEPROM not detected

Ethernet34: SFP EEPROM not detected

Ethernet35: SFP EEPROM not detected

Ethernet36: SFP EEPROM not detected

Ethernet37: SFP EEPROM not detected

Ethernet38: SFP EEPROM not detected

Ethernet39: SFP EEPROM not detected

Ethernet40: SFP EEPROM not detected

Ethernet41: SFP EEPROM not detected

Ethernet42: SFP EEPROM not detected

Ethernet43: SFP EEPROM not detected

Ethernet44: SFP EEPROM not detected

Ethernet45: SFP EEPROM not detected

Ethernet46: SFP EEPROM not detected

Ethernet47: SFP EEPROM not detected

Ethernet48: SFP EEPROM detected
        Connector: LC
        Encoding: NRZ
        Extended Identifier: Power Class 4(3.5W max), CDR present in Rx Tx
        Extended RateSelect Compliance: QSFP+ Rate Select Version 1
        Identifier: QSFP28
        Length(km): 2
        Nominal Bit Rate(100Mbs): 255
        Specification compliance:
        Vendor Date Code(YYYY-MM-DD Lot): 2018-09-13
        Vendor Name: CIG
        Vendor OUI: 00-00-00
        Vendor PN: EA100WDM02AFB09
        Vendor Rev: 18
        Vendor SN: EAA1827000000027


Ethernet49: SFP EEPROM detected
        Connector: LC
        Encoding: NRZ
        Extended Identifier: Power Class 4(3.5W max), CDR present in Rx Tx
        Extended RateSelect Compliance: QSFP+ Rate Select Version 1
        Identifier: QSFP28
        Length(km): 2
        Nominal Bit Rate(100Mbs): 255
        Specification compliance:
        Vendor Date Code(YYYY-MM-DD Lot): 2018-09-13 
        Vendor Name: CIG
        Vendor OUI: 00-00-00
        Vendor PN: EA100WDM02AFB01
        Vendor Rev: 18
        Vendor SN: EAA1827000000016

Ethernet50: SFP EEPROM not detected

Ethernet51: SFP EEPROM not detected

Ethernet52: SFP EEPROM not detected

Ethernet53: SFP EEPROM not detected

Ethernet54: SFP EEPROM not detected

Ethernet55: SFP EEPROM not detected


root@switch1:~# 
root@switch1:~# 
root@switch1:~# show platform syseeprom
TlvInfo Header:
   Id String:    TlvInfo
   Version:      1
   Total Length: 165
TLV Name             Code Len Value
-------------------- ---- --- -----
Product Name         0x21  14 CIG-CS6436-56P
Part Number          0x22  13 FP4ZZ7656749A
Serial Number        0x23  14 CS6436X1830902
Base MAC Address     0x24   6 EC:84:B4:E6:12:B7
Manufacture Date     0x25  19 02/24/2019 13:52:10
Device Version       0x26   1 1
Label Revision       0x27   3 AM2
Platform Name        0x28  24 x86_64-cig_cs6436_56p-r0
ONIE Version         0x29  13 2017.05.00.08
MAC Addresses        0x2A   2 16
Manufacturer         0x2B   3 CIG
Manufacture Country  0x2C   2 CN
Vendor Name          0x2D   9 Cambridge
Diag Version         0x2E   8 0.0.0.13
CRC-32               0xFE   4 0xDB29CAA6
(checksum valid)
root@switch1:~# 
root@switch1:~# show platform psustatus 
PSU    Status
-----  --------
PSU 1  OK
PSU 2  OK
root@switch1:~# 
root@switch1:~# show platform summary 
Platform: x86_64-cig_cs6436_56p-r0
HwSKU: Cig-CS6436-56P
ASIC: nephos
root@switch1:~#

**Comment by michealylj1** [comment]
Already merged into 201811 branch. So just close it.

**Pull request closed without merging**